test/arrayops.jl

```diff
@@ -1106,6 +1106,9 @@ end
     @test isequal(intersect([1,2,3], Float64[]), Float64[])
     @test isequal(intersect(Int64[], [1,2,3]), Int64[])
     @test isequal(intersect(Int64[]), Int64[])
+    @test isequal(intersect([1, 3], 1:typemax(Int)), [1, 3])
+    @test isequal(intersect(1:typemax(Int), [1, 3]), [1, 3])
+
```

Comment:
Can we add a test or 2 for mixed types?

Reply:
Yes, check b77fc1c91211a16f204b47b4acd98b15e044c2ec please